msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/ui/PrivateMessageManagerImpl.java

```diff
@@ -1914,4 +2040,225 @@ else if (toolId.equals(DiscussionForumService.MESSAGES_TOOL_ID))
+
+		  sendPrivateMessage(rrepMsg, recipients, false);
+
+		  if (!rrepMsg.getDraft()) {
+			  markMessageAsRepliedForUser(currentMessage, rrepMsg.getAuthorId());
+			  String contextId = ((PrivateMessageRecipientImpl)currentMessage.getRecipients().get(0)).getContextId();
+			  markMessageAsReadForUser(currentMessage, contextId, rrepMsg.getAuthorId(), DiscussionForumService.MESSAGES_TOOL_ID);
+			  LRS_Statement statement = null;
+			  if (null != learningResourceStoreService) {
+				  try{
+					  statement = getStatementForUserSentPvtMsg(currentMessage.getTitle(), SAKAI_VERB.responded, currentMessage);
+				  }catch(Exception e){
+					  log.error(e.getMessage(), e);
+				  }
+			  }
+
+			  Event event = eventTrackingService.newEvent(DiscussionForumService.EVENT_MESSAGES_RESPONSE, getEventMessage(currentMessage, DiscussionForumService.MESSAGES_TOOL_ID, rrepMsg.getAuthorId(), ((PrivateMessageRecipientImpl)currentMessage.getRecipients().get(0)).getContextId()), null, true, NotificationService.NOTI_OPTIONAL, statement);
+			  eventTrackingService.post(event);
+		  }
+	  }
+  }
+  
+  private static String encrypt(String value) {
+	  try {
+		  IvParameterSpec iv = new IvParameterSpec(INIT_VECTOR.getBytes(StandardCharsets.UTF_8));
+		  String key = ServerConfigurationService.getString("msgcntr.no.reply.secret", "1111111111111111");
+		  SecretKeySpec skeySpec = new SecretKeySpec(key.getBytes(StandardCharsets.UTF_8), "AES");
+
+		  Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5PADDING");
+		  cipher.init(Cipher.ENCRYPT_MODE, skeySpec, iv);
+
+		  byte[] encrypted = cipher.doFinal(value.getBytes(StandardCharsets.UTF_8));
+
+		  return Base64.encodeBase64String(encrypted);
+
+	  } catch (Exception ex) {
+		  log.error(ex.getMessage(), ex);
+	  }
+
+	  return null;
+  }
+
+  private static String decrypt(String encrypted) {
+	  try {
+		  IvParameterSpec iv = new IvParameterSpec(INIT_VECTOR.getBytes(StandardCharsets.UTF_8));
+		  String key = ServerConfigurationService.getString("msgcntr.no.reply.secret", "1111111111111111");
+		  SecretKeySpec skeySpec = new SecretKeySpec(key.getBytes(StandardCharsets.UTF_8), "AES");
+
+		  Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5PADDING");
+		  cipher.init(Cipher.DECRYPT_MODE, skeySpec, iv);
+
+		  byte[] original = cipher.doFinal(Base64.decodeBase64(encrypted));
+
+		  return new String(original,StandardCharsets.UTF_8);
+	  } catch (Exception ex) {
+		  log.error(ex.getMessage(), ex);
+	  }
+
+	  return null;
+  }
+  
+  private LRS_Statement getStatementForUserSentPvtMsg(String subject, SAKAI_VERB sakaiVerb, PrivateMessage rrepMsg) {
+	  LRS_Actor student = learningResourceStoreService.getActor(rrepMsg.getCreatedBy());
+	  String url = ServerConfigurationService.getPortalUrl();
+	  LRS_Verb verb = new LRS_Verb(sakaiVerb);
+	  LRS_Object lrsObject = new LRS_Object(url + "/privateMessage", "send-private-message");
+	  HashMap<String, String> nameMap = new HashMap<>();
```

Comment:
use interface type `Map<String, String>`

Reply:
This implementation has been covered by /messageforums-tool/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java and nothing has been modified from the original